client/src/main/java/com/microsoft/durabletask/TaskOrchestrationExecutor.java

```diff
@@ -289,6 +294,17 @@ public void continueAsNew(Object input, boolean preserveUnprocessedEvents) {
             this.continuedAsNew = true;
             this.continuedAsNewInput = input;
             this.preserveUnprocessedEvents = preserveUnprocessedEvents;
+
+            // If there are events haven't been processed, we make sure to finish processing them before the interruption,
+            // so no event is dropped.
+            while(processNextEvent()) {}
+
+            // The ContinueAsNewInterruption exception allows the orchestration to complete immediately and return back
+            // to azure functions durable extension/sidecar.
```

Comment:
I recommend just calling this a sidecar. Whether it's Azure Functions or Dapr Workflow or the .NET sidecar, it can still be considered a "sidecar". I worry that talking about Azure Functions in a non-Azure Functions project could confuse people.
```suggestion
            // to the sidecar.
```